File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        vector<int> ans(2, 0);
        int l = 0;
        int r = numbers.size() - 1;

        while (r > l) {
            if (numbers[r] + numbers[l] > target) {
                --r;
            } else if (numbers[r] + numbers[l] < target) {
                ++l;
            } else {
                ans[0] = l + 1;
                ans[1] = r + 1;
                break;
            }

        }
        return ans;
    }
};

File: Data Structures & Algorithms/two-integer-sum/submission-2.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        check = {}

        for i, x in enumerate(nums):
            y = target - x

            if y in check:
                return [check[y], i]

            check[x] = i

        return []